### PR TITLE
add review reminder option for member of role

### PR DESCRIPTION
**Pull request by noy93845**

### CODE REVIEW

File: athenz/utils.go [hunk 1]
```diff
@@ -350,3 +359,16 @@ func validateExpirationPatternFunc(validPattern string, attribute string) schema
...
+		r, e := regexp.Compile(validPattern)
+		if e != nil {
+			return diag.FromErr(e)
+		}
+		if r.FindString(val.(string)) != val.(string) {
+			return diag.FromErr(fmt.Errorf("%s must match the pattern %s", attribute, validPattern))
+		}
+		return nil
+	}
+}
```
Comment (mendi160) [inline athenz/utils.go:374]: please don't duplicate the internal func, use the same code for both expiration and review reminder

**Comment by mendi160** [comment]
please update the documentation as well

File: athenz/utils.go (hunk 1 above)
Comment (noy93845) [reply athenz/utils.go]: ok

**Comment by noy93845** [comment]
I updated already the docs

File: athenz/resource_role.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func ResourceRole() *schema.Resource {
 							Type:             schema.TypeString,
 							Optional:         true,
 							Default:          "",
-							ValidateDiagFunc: validateReviewReminderPatternFunc(REVIEW_REMINDER_PATTERN, MEMBER_REVIEW_REMINDER),
+							ValidateDiagFunc: validateExpirationPatternFunc(REVIEW_REMINDER_PATTERN, MEMBER_REVIEW_REMINDER),
```
Comment (mendi160) [inline athenz/resource_role.go:74]: please make the func name more generic

Comment (noy93845) [reply athenz/resource_role.go]: done

**Review by mendi160 — APPROVED** [review]

**Review by mendi160 — DISMISSED** [review]

File: athenz/consts.go [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
+	POLICY_SEPARATOR        = ":policy."
+	RESOURCE_SEPARATOR      = ":"
+	SERVICE_SEPARATOR       = "."
+	SUB_DOMAIN_SEPARATOR    = "."
+	PREFIX_USER_DOMAIN      = "home."
+	EXPIRATION_LAYOUT       = "2006-01-02 15:04:05"
+	MEMBER_EXPIRATION       = "member expiration"
+	EXPIRATION_PATTERN      = "[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9] [0-9][0-9]:[0-9][0-9]:[0-9][0-9]"
+	MEMBER_REVIEW_REMINDER  = "member review reminder"
+	REVIEW_REMINDER_PATTERN = "[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9] [0-9][0-9]:[0-9][0-9]:[0-9][0-9]"
```
Comment (dvirguttman) [inline athenz/consts.go:16]: use the same constant and rename it to date pattern maybe?

File: athenz/resource_role_test.go [hunk 4]
```diff
@@ -757,7 +761,7 @@ resource "athenz_role" "roleTest" {
 `, name, domain, member1)
 }
 
-func testAccGroupRoleConfigAddMemberWithExpiration(name, domain, member1, member2 string) string {
+func testAccGroupRoleConfigAddMemberWithExpirationAndReview(name, domain, member1, member2 string) string {
```
Comment (dvirguttman) [inline athenz/resource_role_test.go:764]: is it possible to have a review without expiration?
why to  test both together?

**Comment by dvirguttman** [comment]
General comment:
Lets move all role settings into settings structure:
```
resource "athenz_role" "role" {
  name = "foo"
  domain = "bar"
  settings = {
    setting1 = ""
    ...
  }
  member {
    name = "m1"

  }
  member {
    name = "m2"
  }
}
```

File: athenz/consts.go (hunk 3 above)
Comment (noy93845) [reply athenz/consts.go]: Done, I renamed it

File: athenz/resource_role_test.go (hunk 4 above)
Comment (noy93845) [reply athenz/resource_role_test.go]: Done,  now its seperated

**Review by dvirguttman — APPROVED** [review]

**Pull request merged**